### PR TITLE
Add source management and test stubs

### PR DESCRIPTION
**Pull request by abaptiste**
Happy New Year! :)

Changes in this PR include:

- Stub out the source tree and source manager modules and their tests.
- Fix a few instances in a BUILD where `srcs` are labeled as `data`
- Fix the comments in `constants.py` to comply with pylint

Signed-off-by: abaptiste <abaptiste@users.noreply.github.com>

cc: @landesherr, @mum4k

### CODE REVIEW

File: salvo/src/lib/source_tree.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
+"""This module contains methods for managing a source tree"""
+
+import logging
+import os
+import re
+from subprocess import CalledProcessError
```
Comment (mum4k) [inline salvo/src/lib/source_tree.py:6]: Can we import `subprocess` instead to follow the guidance of importing packages and modules only?

https://github.com/envoyproxy/envoy-perf/blob/9fd6a9b18943750bb674b2543210559e5c9fe58c/salvo/src/lib/docker/test_docker_image.py#L11-L12

Note - import from the typing module are specifically listed as an exception to this rule.

**Review by mum4k — CHANGES_REQUESTED** [review]
A happy new your to you as well. Sorry about the delay here, ready to get started now.

File: salvo/src/lib/source_manager.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import src.lib.source_tree as source_tree
+import api.source_pb2 as proto_source
+import api.control_pb2 as proto_control
+
+log = logging.getLogger(__name__)
+
+"""The KNOWN_REPOSITORIES map contains the known remote locations for the source
+   code needed to build Envoy.
+"""
+KNOWN_REPOSITORIES = {'envoy': 'https://github.com/envoyproxy/envoy.git'}
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_manager.py:17]: Certainly.  Will update

File: salvo/src/lib/source_manager.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+"""
+This module abstracts the higher level functions of managing source
+code
+"""
+import logging
+from typing import List
+
+import src.lib.source_tree as source_tree
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_manager.py:8]: Right.  I've been trying to change these as I find them.

File: salvo/src/lib/source_manager.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        minimum.  We will not build those from source yet.
+    """
+    return []
+
+  def get_envoy_hashes_for_benchmark(self) -> List[str]:
+    """Determine the hashes for the baseline and previous Envoy Image.
+
+    Using the name and tag for the envoy image specified in the control
+    document, determine the previous image hash.  Return all discovered
+    hashes in a dictionary.
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_manager.py:92]: I will update.  This was overlooked

File: salvo/src/lib/source_manager.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    Returns:
+      a List of commit hashes discovered.
+
+    Raises:
+      SourceManagerError: if we are not able to deterimine hashes prior to
+        the identified commit
+    """
+    return []
+
+  def get_source_repository(self, source_id) -> proto_source.SourceRepository:
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_manager.py:123]: Updated, with examples.

File: salvo/src/lib/source_tree.py (hunk 1 above)
Comment (abaptiste) [reply salvo/src/lib/source_tree.py:6]: Sure

File: salvo/src/lib/source_tree.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+import src.lib.cmd_exec as cmd_exec
+import src.lib.constants as constants
+
+import api.source_pb2 as proto_source
+
+log = logging.getLogger(__name__)
+
+
+TAG_REGEX = r'^v\d\.\d+\.\d+'
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_tree.py:17]: Updating.

[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_tree.py:17]: Certainly. Added a comment

File: salvo/src/lib/source_tree.py [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        os.mkdir(self._tempdir)
+
+      self._source_path = self._tempdir
+      return True
+
+    # We have a working directory on disk and can deduce the origin from it
+    if self._source_path and not self._source_url:
+      return True
+
+    return False
```
[the remark this answers is not in the document]
Comment (abaptiste) [reply salvo/src/lib/source_tree.py:91]: I will change it as you suggested so that we raise an error with a message

**Comment by abaptiste** [comment]
@mum4k This is ready for another pass from you.  Thanks!

**Review by mum4k — APPROVED** [review]
Looks good, thank you.

**Comment by abaptiste** [comment]
Thanks for the merge @mum4k

**Pull request merged**